modules/dnn/src/layers/fast_convolution/depthwise_convolution.cpp

```diff
@@ -84,19 +84,20 @@ static void depthWiseBlockConv2D(const float* wptr,
         v_float32x4 vbias = v_setall_f32(bias);
         v_float32x4 vrc = v_setall_f32(relu_coeff);
 
-        if (stride_w == 1 || (stride_w == 2 && dilation_w == 1))
+        if ((stride_w == 1 || (stride_w == 2 && dilation_w == 1)) && outW1 >= VEC_NLANES)
```

Comment:
Another problem of this optimization.

Line 72:

```
outW1 = max(out_j, outW1 - outW1%VEC_NLANES);
```

This code is under `#if CV_SIMD` but its result affects behavior of "scalar" code below (Line 155):

```
#endif

        for (; out_j < outW1; out_j++)
```

(at the same time we could execute or not execute of any SIMD code)